app/lib/features/chat/widgets/custom_input.dart

```diff
@@ -44,20 +44,25 @@ final _allowEdit = StateProvider.family<bool, String>(
   ),
 );
 
+final canSendProvider = FutureProvider.family<bool?, String>(
+  (ref, roomId) async {
+    final membership = ref.watch(roomMembershipProvider(roomId));
+    return membership.valueOrNull?.canString('CanSendChatMessages');
+  },
+);
+
 class CustomChatInput extends ConsumerWidget {
+  static const noAccessKey = Key('custom-chat-no-access');
```

Comment:
I'm okay with having keys here for now but for future reference, let's keep these under separate files where it's easier to read up and link ;)

Reply:
I disagree, having them as static on the main Objects makes it immediately clear where they are used to be for and nicely namespaces them. We had them in other files and they started to rod, meaning that we collected Keys that were never used and didn't serve any purpose. This way we keep them close to where they are used and through the classname their use is totally clear. I'd prefer to do it this way.